include/flatbuffers/flexbuffers.h

```diff
@@ -377,12 +379,44 @@ class Reference {
       case TYPE_NULL: return 0;
       case TYPE_STRING: return flatbuffers::StringToInt(AsString().c_str());
       case TYPE_VECTOR: return static_cast<int64_t>(AsVector().size());
+      case TYPE_BOOL: return static_cast<int64_t>(AsBool());
       default:
       // Convert other things to int.
       return 0;
     }
   }
 
+  bool AsBool() const {
+    if (type_ == TYPE_BOOL) {
```

Comment:
(sorry for the late reply, somehow missed this)
Yeah, this is indeed tricky. I guess one could argue that if someone writes `"true"` instead of `true` then maybe that is what they want, or if they didn't, then the fact that they get costlier string storage is their own fault.
If we go that way though, you could also argue that a string should always be `false`, regardless of wether it contains `"true"`, since a string is not a boolean. Or maybe, `""` should be false and all other strings true, analogous to how it converts strings/vectors to int. That is already implied by `return AsInt64() != 0`. I guess I don't like this string special case.
If were to go with having the parser convert `"true"` to a boolean value, then `ToString` would get the correct string back, but `AsString` wouldn't. So yeah, maybe lets not go with that.
I'm kind of in favour of the simplest solution here, meaning no special case at all.